codec-http/src/main/java/io/netty/handler/codec/http/HttpDecoderConfig.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+import static io.netty.util.internal.ObjectUtil.checkPositive;
+
+/**
+ * A configuration object for specifying the behaviour of {@link HttpObjectDecoder} and its subclasses.
+ * <p>
+ * The {@link HttpDecoderConfig} objects are mutable to reduce allocation,
```

Comment:
Mutability also has problems. WDYT about introducing `HttpDecoderConfig` interface and `HttpDecoderConfigBuilder` as a final class? Builder will be a good compromise between memory allocations and providing an immutable config by default. For cases when users need runtime mutability, they can implement interface.

Reply:
The constructors that take `HttpDecoderConfig` objects always extract all fields, so I think it's okay.